pandas/tests/scalar/timedelta/test_arithmetic.py

```diff
@@ -200,6 +200,80 @@ def test_td_rsub_numeric_raises(self):
         with pytest.raises(TypeError):
             2.0 - td
 
+    def test_td_sub_timedelta_timedeltalike_array(self):
+        # GH 21980
+        arr = np.array([Timestamp('20130101 9:01'),
+                        Timestamp('20121230 9:02')])
+        exp = np.array([Timestamp('20121231 9:01'),
+                        Timestamp('20121229 9:02')])
+        res = arr - pd.Timedelta('1D')
+        tm.assert_numpy_array_equal(res, exp)
+
+    def test_td_sub_timedelta_mixed_timedeltalike_array(self):
+        # GH 21980
+        now = pd.Timestamp.now()
+        arr = np.array([now,
+                        pd.Timedelta('1D')])
+        exp = np.array([now - pd.Timedelta('1D'),
+                        pd.Timedelta('0D')])
+        res = arr - pd.Timedelta('1D')
+        tm.assert_numpy_array_equal(res, exp)
+
+    def test_td_rsub_timedelta_timedeltalike_array(self):
+        # GH 21980
+        arr = np.array([Timestamp('20130101 9:01'),
+                        Timestamp('20121230 9:02')])
+        with pytest.raises(TypeError):
+            # an timedelta - timestamp doesnt make sense
+            pd.Timedelta('1D') - arr
+
+    def test_td_rsub_timedelta_mixed_timedeltalike_array(self):
+        # GH 21980
+        now = pd.Timestamp.now()
+        arr = np.array([now,
+                        pd.Timedelta('1D')])
+        with pytest.raises(TypeError):
+            # an timedelta - timestamp doesnt make sense
+            pd.Timedelta('1D') - arr
+
+    def test_td_add_timedelta_timedeltalike_array(self):
+        # GH 21980
+        arr = np.array([Timestamp('20130101 9:01'),
+                        Timestamp('20121230 9:02')])
+        exp = np.array([Timestamp('20130102 9:01'),
+                        Timestamp('20121231 9:02')])
+        res = arr + pd.Timedelta('1D')
+        tm.assert_numpy_array_equal(res, exp)
+
+    def test_td_add_timedelta_mixed_timedeltalike_array(self):
+        # GH 21980
+        now = pd.Timestamp.now()
+        arr = np.array([now,
+                        pd.Timedelta('1D')])
+        exp = np.array([now + pd.Timedelta('1D'),
+                        pd.Timedelta('2D')])
+        res = arr + pd.Timedelta('1D')
+        tm.assert_numpy_array_equal(res, exp)
+
+    def test_td_radd_timedelta_timedeltalike_array(self):
+        # GH 21980
+        arr = np.array([Timestamp('20130101 9:01'),
+                        Timestamp('20121230 9:02')])
+        exp = np.array([Timestamp('20130102 9:01'),
+                        Timestamp('20121231 9:02')])
+        res = pd.Timedelta('1D') + arr
+        tm.assert_numpy_array_equal(res, exp)
+
+    def test_td_radd_timedelta_mixed_timedeltalike_array(self):
```

Comment:
The "timedelta" in "radd_timedelta" is redundant, since that is what the "td" in "td_radd" stands for.

Same comments as below re naming and parametrizing.

Reply:
Done